spicy/toolchain/src/compiler/visitors/validator.cc

```diff
@@ -416,6 +416,14 @@ struct PreTransformVisitor : public hilti::visitor::PreOrder<void, PreTransformV
 
         if ( auto contexts = u.propertyItems("%context"); contexts.size() > 1 )
             error("unit cannot have more than one %context", p);
+
+        if ( auto&& typeId = u.typeID() ) {
```

Comment:
Style nit: prefer `const X&` over `X &&` for const access, for readability. 

```suggestion
        if ( const auto& typeId = u.typeID() ) {
```